### PR TITLE
v8 regression invalidates `superFunction`'s primer

### PR DESCRIPTION
**Pull request by jasonmit**
This takes on the same strategy of https://github.com/jasonmit/ember.js/blob/master/packages/ember-runtime/lib/system/core_object.js#L179-L191
- [x] Add back args length 2 condition
- [x] Fix failing recursive super call test

/cc @stefanpenner

### CODE REVIEW

File: packages/ember-metal/lib/mixin.js [hunk 1]
```diff
@@ -48,37 +48,28 @@ var REQUIRED;
...
-      ret = func.call(this, arguments[0]);
-    } else if (length === 2) {
-      ret = func.call(this, arguments[0], arguments[1]);
-    } else {
-      ret = func.apply(this, arguments);
+  
+  if (length === 0) {
+    ret = func.call(this);
+  } else if (length === 1) {
+    ret = func.call(this, arguments[0]);
```
Comment (stefanpenner) [inline packages/ember-metal/lib/mixin.js]: we lost the length === 2

**Comment by jasonmit** [comment]
@stefanpenner tests are passing now

**Comment by rwjblue** [comment]
@stefanpenner - Needs review.

**Comment by rwjblue** [comment]
@jasonmit - Needs a rebase.

@stefanpenner - Needs review (might want to do a general review before @jasonmit does a rebase)...

**Comment by rwjblue** [comment]
I'm going to close this because I think the recent work in https://github.com/emberjs/ember.js/pull/12062 and https://github.com/emberjs/ember.js/pull/12132 superceded it.

**Pull request closed without merging**